Dockerfile

```diff
@@ -1,39 +1,16 @@
-# To use or update to a ruby version, change {BASE_RUBY_IMAGE}
+## To use or update to a ruby version, change {BASE_RUBY_IMAGE}
 ARG BASE_RUBY_IMAGE=ruby:2.7.4-alpine3.12
-# BASE_RUBY_IMAGE_WITH_GEMS_AND_NODE_MODULES will default to apply-for-teacher-training-gems-node-modules
-# building all layers above it if a value is not specidied during the build
-ARG BASE_RUBY_IMAGE_WITH_GEMS_AND_NODE_MODULES=apply-for-teacher-training-gems-node-modules
 
-# Stage 1: install-gems-node-modules, download gems and node modules.
-FROM ${BASE_RUBY_IMAGE} AS install-gems-node-modules
+# Stage 1: gems-node-modules, build gems and node modules.
+FROM ${BASE_RUBY_IMAGE} AS gems-node-modules
 
-ARG BUILD_DEPS="git gcc libc-dev make nodejs yarn postgresql-dev build-base libxml2-dev libxslt-dev ttf-ubuntu-font-family"
-ENV WKHTMLTOPDF_GEM=wkhtmltopdf-binary-edge-alpine
-
-WORKDIR /app
-
-COPY Gemfile Gemfile.lock package.json yarn.lock ./
+ARG BUILD_DEPS="git gcc libc-dev make postgresql-dev build-base libxml2-dev libxslt-dev ttf-ubuntu-font-family"
 
 RUN apk -U upgrade && \
     apk add --update --no-cache --virtual .gem-installdeps $BUILD_DEPS && \
-    gem update --system && \
-    find / -wholename '*default/bundler-*.gemspec' -delete && \
-    rm -rf /usr/local/bin/bundle && \
-    gem install bundler -v 2.1.4 && \
-    bundler -v && \
-    bundle config set no-cache 'true' && \
-    bundle config set no-binstubs 'true' && \
-    bundle --retry=5 --jobs=4 --without=development --with=production && \
-    yarn install --check-files && \
-    apk del .gem-installdeps && \
-    rm -rf /usr/local/bundle/cache && \
-    find /usr/local/bundle/gems -name "*.c" -delete && \
-    find /usr/local/bundle/gems -name "*.h" -delete && \
-    find /usr/local/bundle/gems -name "*.o" -delete
-
-# Stage 2: apply-for-teacher-training-gems-node-modules, reduce size of gems-node-modules and only keep required files.
-# published as dfedigital/apply-for-teacher-training-gems-node-modules
-FROM ${BASE_RUBY_IMAGE} AS apply-for-teacher-training-gems-node-modules
+    apk add --update --no-cache nodejs yarn tzdata libpq libxml2 libxslt graphviz && \
```

Comment:
nodejs and yarn could be in BUILD_DEPS

Reply:
since we don't really need to optimise this image for size `--virtual .gem-installdeps` and `apk del .gem-installdeps` may be removed

Reply:
BUILD_DEPS gets removed after bundle install. but I want to keep node and yarn as part of final image as its required for running yarn test on the gems-node-modules image later in the pipeline.

Reply:
This big RUN command can be split into mutiple RUNs for readability since they will be cached independently